internal/clients/ec2/ec2_errors.go

```diff
@@ -0,0 +1,11 @@
+package ec2
+
+import (
+	"fmt"
+)
+
+var OperationNotPermit = fmt.Errorf("operation not permitted")
+
+func OperationNotAllowedError(op string) error {
+	return fmt.Errorf("OperationNotPermit %w : %s", OperationNotPermit, op)
+}
```

Comment:
The name for this error imply that this was an operation that was not permitted. Looks like this is a feature that hasn't been implemented to me. But this is okay.

What I would like us to use is Err suffix for all error values, which is the standard practice in Go library. This would become `OperationNotPermittedErr`. On top of that, you must not use `fmt.Errorf` to create the error because then you will have the linter complaining about this. All it wants you to do is this:

```go
OperationNotPermittedErr := errors.New("operation not permitted")
```

Then simply return this value: `return nil, OperationnotPermittedErr` it is as easy as that. The caller simply can compare the error: `if err == OperationNotPermittedErr` too. Errors are values (Go proverb), I love that :-)

Reply:
Do you mean that there is no need for that function? 
I created it because of the linter complaints, that way there are no warnings from the linter. 

do you mean to change it to: 
```golang
OperationNotPermittedErr := errors.New("operation not permitted")
	if len(resp.InstanceTypes) == 100 {
		return nil, OperationNotPermittedErr
	}
```

because that way I still get a complaint from the linter: 
```golang
do not define dynamic errors, use wrapped static errors instead: "errors.New(\"operation not permitted\")"
```

Reply:
So I am not sure when is it correct to use fmt.Errorf and when should I used errors.New?

Reply:
So errors.New is when you want to create new variable (value) that is the "source" error. In Go, error is an interface that implements Error function and that is all. It can be any value or type, but what is very typical in Go is that you simply use plain strings as error types. And this is what errors.New returns, implementation of error that is just a string. They are typically all lowercase (no sentence) in the Go standard library. An example:

https://cs.opensource.google/go/go/+/master:src/internal/oserror/errors.go

In Go 1.13 a new Wrapper interface was introduced which allows to "wrap" errors into errors which is very useful. Typically, an API should always return its own errors wrapping underlying errors like "file not found". The concept is really easy, all you need to do is to implement Unwrap function for your own error. Similarly like errors.New which returns a simple string, there is a slightly more complex wrapped string which can be created with several ways, one of it is fmt.Errorf which returns such implementation.

To know more how to actually work with wrapped errors I recommend: https://go.dev/blog/go1.13-errors

Reply:
Thanks!